### ISSUE TITLE
MSL & GLSL: OpBranchConditional throws nullptr error from the True Label

### ISSUE THREAD

**Issue by billhollings** [KhronosGroup/SPIRV-Cross#1768]
These CTS test fail after SPIR-V conversion to MSL throws `nullptr` error from the _True Label_:

```
dEQP-VK.spirv_assembly.type.scalar.i64.switch_vert
dEQP-VK.spirv_assembly.type.scalar.i64.switch_tessc
dEQP-VK.spirv_assembly.type.scalar.i64.switch_tesse
dEQP-VK.spirv_assembly.type.scalar.i64.switch_frag
dEQP-VK.spirv_assembly.type.scalar.i64.switch_comp
dEQP-VK.spirv_assembly.type.scalar.u64.switch_vert
dEQP-VK.spirv_assembly.type.scalar.u64.switch_tessc
dEQP-VK.spirv_assembly.type.scalar.u64.switch_tesse
dEQP-VK.spirv_assembly.type.scalar.u64.switch_frag
dEQP-VK.spirv_assembly.type.scalar.u64.switch_comp
```

The related conversion to GLSL also fails.

SPIR-V files: 
[nullptr_on_branch_conditional_true_label.spv.zip](https://github.com/KhronosGroup/SPIRV-Cross/files/7341037/nullptr_on_branch_conditional_true_label.spv.zip)

**Comment by billhollings** [comment]
Unfortunately, PR #1794 did not fix this issue.

The error encountered is now the assertion throw: "Cannot use 64-bit switch selectors."

To explore this, I bypassed this  assertion, and tweaked this line to allow the switch statement to be created:

```
bool degenerate_switch = block.default_block != block.merge_block && block.cases_32bit.empty() && block.cases_64bit.empty();
```

However, the following SPIR-V:

```
  %88 = OpTypeInt 64 1
 %118 = OpLoad %88 %117
        OpSwitch %118 %122 3210 %123 -268438669 %124 26843866939192872 %125
```

specifies the literals as 64-bit values.

The problem seems to be this code, that assumes only 32-bit literals in the case statements:

https://github.com/KhronosGroup/SPIRV-Cross/blob/37dfb3f45f4fc47c841f81e618c602f6f3de0f17/spirv_glsl.cpp#L14991-L14992

**Comment by etra0** [comment]
I was about to re-open this issue tomorrow actually haha, but yeah, the PR actually fixes a segfault when the switch cases are 64 bit width and negative (because it would try to read a variable at the position 0xFFFF FFFF. To fix this, as @HansKristian-Work suggested, the OpSwitch with 64 bit wide conditions should be rewritten as nested ifs I think, because from my understanding, GLSL doesn't support 64 bit integers in switch statement (I tried with glslc and reading the spec).

**Comment by billhollings** [comment]
>GLSL doesn't support 64 bit integers in switch statement

Hmmm...I expect that MSL does, however, since it is a subset of C++14. Perhaps we can refactor to accommodate that?

**Comment by billhollings** [comment]
Fixed by #1815.
